tinodesdk/src/main/java/co/tinode/tinodesdk/Tinode.java

```diff
@@ -806,9 +829,10 @@ public void oobNotification(Map<String, String> data, String authToken, boolean
                         } catch (Exception ignored) {}
                     }
 
-                    if (newConnection && !keepConnection) {
-                        disconnect(true);
+                    if (keepConnection) {
+                        pinConnectionToFg();
                     }
+                    disconnect(true);
```

Comment:
The name is misleading: it sounds imperative while in reality the disconnect is conditional.
Perhaps, rename `disconnectIfNeeded` or something like this?

Reply:
done: renamed `maybeDisconnect`